### PR TITLE
add tests for generateTexTree

### PR DESCRIPTION
**Pull request by dehann**
remains an experimental feature, that requires `dot2tex`

cc @tonioteran -- thanks for contributing!

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -74,6 +74,8 @@ end
 # include("testExpandedJLD.jl")
 
 
+include("testTexTreeIllustration.jl")
```
Comment (tonioteran) [inline test/runtests.jl]: where do we add these files?

Comment (dehann) [reply test/runtests.jl]: they are alongside in the same folder

**Comment by dehann** [comment]
test not passing yet, although Travis thinks so. The underlying latex issue is a silent error, but can see in log.

**Comment by dehann** [comment]
Hi @tonioteran , do you perhaps know which latex packages are needed for this?  I'm trying to avoid installing all of `texlive` every time.  The current packages being installed are:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/d5fe273cdb56edc626bb08b10ca701d7f9be5155/.travis.yml#L24-L29

Current packages that are installed from a log:
```bash
Recommended packages:
  pgf | texlive-pstricks lmodern texlive-latex-base-doc
The following NEW packages will be installed:
  dot2tex graphviz hdf5-tools libaec0 libcupsfilters1 libcupsimage2
  libgfortran3 libgs9 libgs9-common libharfbuzz-icu0 libhdf5-10 libijs-0.35
  libjbig2dec0 libjs-sphinxdoc libjs-underscore libkpathsea6 libpaper-utils
  libpaper1 libpoppler58 libpotrace0 libptexenc1 libsynctex1 libsz2
  libtexlua52 libtexluajit2 libzzip-0-13 poppler-data **preview-latex-style**
  python-pyparsing t1utils tex-common texlive-base texlive-binaries
  **texlive-latex-base**
The following packages will be upgraded:
  libcups2
1 upgraded, 34 newly installed, 0 to remove and 219 not upgraded.
Need to get 41.0 MB of archives.
```

and then the pip install:
```
$ pip install dot2tex
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/__init__.py:83: RequestsDependencyWarning: Old version of cryptography ([1, 2, 3]) may cause slowdown.
  warnings.warn(warning, RequestsDependencyWarning)
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Requirement already satisfied: dot2tex in /usr/lib/python2.7/dist-packages (2.9.0)
You are using pip version 19.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

### The Error

Here is the current [error from Travis](https://travis-ci.org/JuliaRobotics/IncrementalInference.jl/jobs/600471131) when running the Julia tests:
```julia
Bayes Tree
ERROR    No dimension data could be extracted from dot2tex.tex.
ERROR    Failed to preprocess the graph.

Is the preview LaTeX package installed? ((Debian package preview-latex-style)
To see what happened, run dot2tex with the --debug option.
ERROR    Failed to load file /home/travis/build/JuliaRobotics/IncrementalInference.jl/test/tmp/btproc.dot
┌ Warning: ProcessFailedException(Base.Process[Process(`dot2tex /home/travis/build/JuliaRobotics/IncrementalInference.jl/test/tmp/btproc.dot -o /home/travis/build/JuliaRobotics/IncrementalInference.jl/test/tmp/bt.tex`, ProcessExited(1))])
└ @ IncrementalInference ~/build/JuliaRobotics/IncrementalInference.jl/src/JunctionTree.jl:326
stacktrace() = Base.StackTraces.StackFrame[macro expansion at show.jl:576 [inlined], #generateTexTree#108(::String, ::typeof(generateTexTree), ::BayesTree) at JunctionTree.jl:327, (::getfield(IncrementalInference, Symbol("#kw##generateTexTree")))(::NamedTuple{(:filepath,),Tuple{String}}, ::typeof(generateTexTree), ::BayesTree) at none:0, top-level scope at testTexTreeIllustration.jl:47, top-level scope at Test.jl:1113, top-level scope at testTexTreeIllustration.jl:14, include at boot.jl:328 [inlined], include_relative(::Module, ::String) at loading.jl:1094, include(::Module, ::String) at Base.jl:31, include(::String) at client.jl:431, top-level scope at runtests.jl:77, include at boot.jl:328 [inlined], include_relative(::Module, ::String) at loading.jl:1094, include(::Module, ::String) at Base.jl:31, include(::String) at client.jl:431, top-level scope at none:5, eval(::Module, ::Any) at boot.jl:330, exec_options(::Base.JLOptions) at client.jl:271, _start() at client.jl:464]
```

**Comment by dehann** [comment]
FYI, can resolve in a week or two if that is better timing.

**Comment by tonioteran** [comment]
I actually do not know which specific ones are needed, but yea, it does seem quite crazy to install the full `texlive` every time. Not sure how to best proceed here.

**Comment by dehann** [comment]
OH snap, think I got it:
```
Bayes Tree
Test Summary:           | Pass  Total
testing generateTexTree |    1      1
   Testing IncrementalInference tests passed 
```

Missing package was `texlive-pictures`, see edba857d

**Comment by dehann** [comment]
Travis is having a little issues with network access today, so don't mind the many failures -- thats happening everywhere.  Will probably be cleared in a day or two.

**Comment by tonioteran** [comment]
nice! looks good man, thanks for taking care of the tests!

**Pull request merged**